lib/pipeline/src/builder.rs

```diff
@@ -65,6 +65,17 @@ impl<Output: Send + 'static> Pipeline<Output> {
         }
     }
 
+    /// Conditionally add a component if present. The component must keep the same item type.
+    pub fn pipe_opt<C>(self, component: Option<C>) -> Pipeline<Output>
+    where
+        C: PipelineComponent<Input = Output, Output = Output>,
+    {
+        match component {
+            Some(c) => self.pipe(c),
+            None => self,
+        }
+    }
```

Comment:
I think it would be cleaner if you wrote a blanket implementation for `Option<C>`. Something like this:
```rust
impl<T: Send + Sync + 'static, C: PipelineComponent<Input = T, Output = T>> PipelineComponent
    for Option<C> {
  // todo: implement
}
```